### PR TITLE
Use nsfw for gulp watch

### PR DESCRIPTION
**Pull request by bpasero**
This makes our file watcher for compilation use the NSFW file watcher (from https://github.com/Axosoft/nsfw) to get more coverage on the quality of file events. There is an environment variable `vscode.disableNewWatcher` to force disable this if needed (thus I did not delete the old code yet).

Most of the code is copied out of the Windows watcher we already had: https://github.com/Microsoft/vscode/blob/ben/nsfw/build/lib/watch/watch-win32.js

The only thing needed was to convert the file events from NSFW to the gulp format that we need (including converting rename events to DELETE and ADD). Though I wonder if we would benefit in any way from knowing that a file was renamed in terms of better recompilation?

/cc @jrieken @alexandrudima @joaomoreno

### CODE REVIEW

File: build/lib/watch/index.js [hunk 1]
```diff
@@ -17,9 +17,13 @@ function handleDeletions() {
 	});
 }
 
-const watch = process.platform === 'win32'
-	? require('./watch-win32')
-	: require('gulp-watch');
+const disableNewWatcher = process.env['vscode.disableNewWatcher'];
```
Comment (Tyriar) [inline build/lib/watch/index.js]: Might be better to not have a `.` in the env variable name, we use `VSCODE_VARNAME` format elsewhere afaik.

**Review by Tyriar — APPROVED** [review]
Looks good 😃

File: build/lib/watch/index.js (hunk 1 above)
Comment (bpasero) [inline build/lib/watch/index.js]: @Tyriar makes sense, pushed an update

**Review by joaomoreno — CHANGES_REQUESTED** [review]
Awesome.

I'd rename `VSCODE_DISABLE_NEW_WATCHER` to `VSCODE_USE_LEGACY_WATCH` and make it mean the opposite.

You also gotta fix `build/npm/preinstall.js#22` and remove the `process.platform !== 'win32'` so we always run `npm install` on the `build/lib/watch` folder, regardless of platform.

**Comment by stevedesmond-ca** [comment]
Heads up, just got this when cross-compiling 1.14.0 for ARM:
```
Could not load our cross platform file watcher: Error: /home/vtsv/vscode-arm/vscode/build/lib/watch/node_modules/nsfw/build/Release/nsfw.node: wrong ELF class: ELFCLASS32
Falling back to our platform specific watcher...
```
I'm assuming the `--arch` flag isn't getting passed to the nsfw build process -- I'll add a proper issue (and PR) this weekend if no one else gets around to it first.

**Pull request merged**